constructor/nsis/main.nsi.tmpl

```diff
@@ -1185,10 +1187,36 @@ Section "Uninstall"
     #   carefully.  More info at https://docs.conda.io/projects/conda/en/latest/user-guide/troubleshooting.html#solution
     System::Call 'kernel32::SetEnvironmentVariable(t,t)i("CONDA_DLL_SEARCH_MODIFICATION_ENABLE", "1").r0'
 
+    # Read variables the uninstaller needs from the registry
```

Comment:
If I understand this correctly, this block will search the registry for two variables:

- INSTALLER_NAME_FULL
- INSTALLER_VERSION

These were previously assumed to be constant and would have used the hardcoded values at install time (`${NAME}` [used in [line 65](https://github.com/conda/constructor/blob/fa2040b199eafe4fb8488df30eeac43f26a951da/constructor/nsis/main.nsi.tmpl#L65)] and `${VERSION}`, respectively).

However we are now saying that this might not be always the case. Could you elaborate under what circumstances this is true? e.g. updating Anaconda in place but not updating the registry keys (or not deleting the old ones before adding the new ones)?

Reply:
I guess this logic allows the uninstaller to run for _other_ conda installations if needed, I don't know if that's a bug or a feature 😬

Reply:
> I guess this logic allows the uninstaller to run for _other_ conda installations if needed, I don't know if that's a bug or a feature 😬

That is already possible. The changes in this PR will, however, conduct checks to make sure the deletion is performed consistently on the file system and the registry.

There is another, admittedly nice, application: if you install the same Miniconda/Miniforge version twice, installation 2 will overwrite the registry keys of number 1. If you then run the uninstaller of number 1, it will delete the keys of number 2.